TiFlash late materialization GA (#13236)
pingcap/docs: tiflash/tiflash-late-materialization.md

# TiFlash Late Materialization

> **Note:**
>

> > TiFlash late materialization is not usable under [Disaggregated Storage and Compute Architecture](/tiflash/tiflash-disaggregated-and-s3.md) and [Fast Scan mode](/tiflash/use-fastscan.md).

Reviewer: ```suggestion
> TiFlash late materialization does not take effect in the [disaggregated storage and compute architecture](/tiflash/tiflash-disaggregated-and-s3.md) and [fast scan mode](/tiflash/use-fastscan.md).
```

# TiFlash Late Materialization

> **Note:**
>
> TiFlash late materialization does not take effect in the [fast scan mode](/tiflash/use-fastscan.md).

TiFlash late materialization is an optimization method to accelerate queries in OLAP scenarios. You can use the [`tidb_opt_enable_late_materialization`](/system-variables.md#tidb_opt_enable_late_materialization-new-in-v700) system variable to control whether to enable or disable TiFlash late materialization.

- When it is disabled, to process a `SELECT` statement with filter conditions (`WHERE` clause), TiFlash reads all the data from the columns required by the query, and then filters and aggregates the data based on the query conditions.
- When it is enabled, TiFlash supports pushing down part of the filter conditions to the TableScan operator. That is, TiFlash first scans the column data related to the filter conditions that are pushed down to the TableScan operator, filters the rows that meet the condition, and then scans the other column data of these rows for further calculation, thereby reducing IO scans and computations of data processing.

> To improve the performance of certain queries in OLAP scenarios, starting from v7.1.0, TiDB enables the TiFlash late materialization feature by default. The TiDB optimizer can determine which filter conditions to be pushed down based on statistics and filter conditions, and prioritize pushing down the filter conditions with high filtration rates. For detailed algorithms, see the [RFC document](https://github.com/pingcap/tidb/tree/master/docs/design/2022-12-06-support-late-materialization.md).

Reviewer: ```suggestion
To improve the performance of certain queries in OLAP scenarios, starting from v7.1.0, the TiFlash late materialization feature is enabled by default. The TiDB optimizer can determine which filter conditions to be pushed down based on statistics and filter conditions, and prioritize pushing down the filter conditions with high filtration rates. For detailed algorithms, see the [RFC document](https://github.com/pingcap/tidb/tree/master/docs/design/2022-12-06-support-late-materialization.md).
```
